tests/e2e/utils/src/flows/with-rest-api.js

```diff
@@ -16,20 +16,24 @@ const userEndpoint = '/wp/v2/users';
 const deleteAllRepositoryObjects = async ( repository, defaultObjectId = null ) => {
 	let objects;
 	const minimum = defaultObjectId == null ? 0 : 1;
+	const statuses = ['publish','trash'];
 
-	objects = await repository.list();
-	while ( objects.length > minimum ) {
-		for (let o = 0; o < objects.length; o++ ) {
-			// Skip default data store object
-			if ( objects[ o ].id == defaultObjectId ) {
-				continue;
+	for ( let s = 0; s < statuses.length; s++ ) {
```

Comment:
Using `statuses.each` here might  back it a bit more readable.

Reply:
> Using statuses.each here might back it a bit more readable.

I had that and was getting the warning that the promises would be ignored and to use the basic for loop instead.

Reply:
Ah! Right. Would need an async `each` then but yes, that doesn't sound enticing.